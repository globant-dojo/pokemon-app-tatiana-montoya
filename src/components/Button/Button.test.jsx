/* External */
import { fireEvent, render, screen } from "@testing-library/react";

/* Componentes */
import { Button } from ".";

describe("Tests on <Button />", () => {
  const children = "Buscar";
  const handleAction = jest.fn();
  const iconName = "icon-search";

  const renderButton = () =>
    render(
      <Button iconName={iconName} handleAction={handleAction}>
        {children}
      </Button>
    );

  test("it should display a span when it receives an iconName", () => {
    const { container } = renderButton();

    expect(container.getElementsByClassName(iconName)).toBeTruthy();
  });

  test("it should display a text when it receives the children", () => {
    renderButton();

    expect(screen.findByText(children)).toBeTruthy();
  });

  test("it should call the onChange when clicked", () => {
    renderButton();

    const button = screen.getByRole("button");

    fireEvent.click(button);

    expect(handleAction).toHaveBeenCalled();
    expect(handleAction).toHaveBeenCalledTimes(1);
  });

  test("it should match with the snapshot", () => {
    const { container } = renderButton();

    expect(container).toMatchSnapshot();
  });
});
