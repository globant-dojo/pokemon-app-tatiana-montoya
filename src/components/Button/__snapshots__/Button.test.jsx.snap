// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Button /> it should match with the snapshot 1`] = `
<div>
  <button
    class="pkm-btn"
    type="button"
  >
    <span
      class="icon-search"
    />
    Buscar
  </button>
</div>
`;
